### PR TITLE
Make ``q`` optional in ``sample``

### PR DESCRIPTION
**Pull request by joecummings**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [x] other (please add here)

User should not have to specify ``q`` in order to use the ``sample`` method. Therefore, we can provide the default. 

TODO: Look into whether setting the RNG is actually necessary or if a global Generator works fine, too.

#### Changelog
* Set default for ``q``

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

```
>>> from torchtune.generation import sample
>>> import torch
>>> compiled_sample = torch.compile(sample, fullgraph=True, backend="aot_eager")
>>> logits = torch.empty(3, 3).uniform_(0, 1)
>>> compiled_sample(logits)
tensor([[1],
        [2],
        [0]], dtype=torch.int32)
>>> sample(logits)
tensor([[1],
        [2],
        [0]], dtype=torch.int32)
```

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [x] I have added an example to docs or docstrings

### CODE REVIEW

File: tests/torchtune/generation/test_generation.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def test_sample_consistency(self):
         q = torch.empty(
             (1, 2000),
         ).exponential_(1)
-        token = sample(logits, q, temperature=1, top_k=1)
+        token = sample(logits, temperature=1, top_k=1, q=q)
```
Comment (pbontrager) [inline tests/torchtune/generation/test_generation.py:227]: Can we remove q here all together?

**Review by pbontrager — APPROVED** [review]
Please test this with all recipes that explicitly use "q". Otherwise looks good.
